Translation of Reference test renderer (#116)
reactjs/pt-BR.reactjs.org: content/docs/reference-test-renderer.md

---
id: test-renderer
title: Renderizador de Teste
permalink: docs/test-renderer.html
layout: docs
category: Reference
---

**Importando**

```javascript
import TestRenderer from 'react-test-renderer'; // ES6
const TestRenderer = require('react-test-renderer'); // ES5 com npm
```

## Visão Geral {#overview}

> Este pacote fornece um renderizador React que pode ser usado para renderizar componentes React para objetos JavaScript puros, sem depender do DOM ou de um ambiente móvel nativo.

Reviewer: ```suggestion
Este pacote fornece um renderizador React que pode ser usado para renderizar componentes do React para objetos JavaScript puros, sem depender do DOM ou de um ambiente móvel nativo.
```

Author: Não é necessário. 
Componente do React implica um componente interno a biblioteca, ao contrário de componente React, que implica que o componente é do 'tipo' React.

* [`testInstance.parent`](#testinstanceparent)
* [`testInstance.children`](#testinstancechildren)

## Referência {#reference}

### `TestRenderer.create()` {#testrenderercreate}

```javascript
TestRenderer.create(element, options);
```

Cria uma instância do `TestRenderer` com o elemento React fornecido. Este método não usa o DOM real, mas ainda renderiza completamente a árvore de componentes na memória para que você possa fazer verificações sobre ela. A instância retornada possui os seguintes métodos e propriedades.

### `testRenderer.toJSON()` {#testrenderertojson}

```javascript
testRenderer.toJSON()
```

> Retorna um objeto representando a árvore renderizada. Esta árvore contém apenas os nós específicos da plataforma como `<div>` ou `<View>` e seus props, mas não contém nenhum componente criado pelo usuário. Isso é útil para [teste de snapshot](http://facebook.github.io/jest/docs/en/snapshot-testing.html#snapshot-testing-with-jest).

Reviewer 2: Percebi que em alguns momentos você usou a tradução _instantâneo_ e em outros usou diretamente _snapshot_
Seria bom manter uniformidade e usar apenas um deles

Reviewer: Acredito que seja melhor usar só snapshot também. Pelo menos pra mim, "recurso de teste de instantâneo" não me dá um contexto do que o Jest está de fato fazendo.

Maas... pode ser só uma impressão minha, aceito segundas opniões 😄

Author: 😅 eu esqueci de alterar todas ocorrências de instantâneo para snapshot. Vou corrigir. Obrigado

> Retorna um objeto representando a árvore renderizada. Esta árvore contém apenas os nós específicos da plataforma como `<div>` ou `<View>` e seus props, mas não contém nenhum componente criado pelo usuário. Isso é útil para [teste de snapshot](http://facebook.github.io/jest/docs/en/snapshot-testing.html#snapshot-testing-with-jest).

Reviewer: ```suggestion
Retorna um objeto representando a árvore renderizada. Esta árvore contém apenas os nós específicos da plataforma como `<div>` ou `<View>` e suas props, mas não contém nenhum componente criado pelo usuário. Isso é útil para [testes de snapshot](http://facebook.github.io/jest/docs/en/snapshot-testing.html#snapshot-testing-with-jest).
```

> Retorna um objeto representando a árvore renderizada. Esta árvore contém apenas os nós específicos da plataforma como `<div>` ou `<View>` e suas props, mas não contém nenhum componente criado pelo usuário. Isso é útil para [testes de snapshot](https://facebook.github.io/jest/docs/en/snapshot-testing.html#snapshot-testing-with-jest).

Reviewer 3: ```suggestion
Retorna um objeto representando a árvore renderizada. Essa árvore contém apenas os nós específicos da plataforma como `<div>` ou `<View>` e suas props, mas não contém nenhum componente criado pelo usuário. Isso é útil para [testes de snapshot](https://facebook.github.io/jest/docs/en/snapshot-testing.html#snapshot-testing-with-jest).
```

### `testRenderer.toTree()` {#testrenderertotree}

```javascript
testRenderer.toTree()
```

Retorna um objeto representando a árvore renderizada. Ao contrário de `toJSON()`, a representação é mais detalhada que a fornecida por `toJSON()`, e inclui os componentes criados pelo usuário. Você provavelmente não precisa desse método, a menos que esteja escrevendo sua própria biblioteca de asserções sobre o renderizador de teste.

### `testRenderer.update()` {#testrendererupdate}

```javascript
testRenderer.update(element)
```

> Re-renderiza a árvore na memória com um novo elemento raiz. Isto simula uma atualização do React na raiz. Se o novo elemento tiver o mesmo tipo e chave do elemento anterior, a árvore será atualizada; caso contrário, ele irá montar novamente uma nova árvore.

Reviewer 3: ```suggestion
Re-renderiza a árvore na memória com um novo elemento raiz. Isso simula uma atualização do React na raiz. Se o novo elemento tiver o mesmo tipo e chave do elemento anterior, a árvore será atualizada; caso contrário, ele irá montar novamente uma nova árvore.
```
